### PR TITLE
docs: update file source docs with details about include/exclude semantics

### PR DESCRIPTION
**Pull request by aroberts**
Currently the File source include/exclude semantics has undocumented behavior that may be unintiutive to some users. Updates the file source docs to include the semantics outlined in #14771.

### CODE REVIEW

**Review by spencergilbert — COMMENTED** [review]
👍 thanks for the change! Though I hate to suggest it, we can probably make the line shorter by doing the same thing as [here](https://github.com/vectordotdev/vector/pull/14776/files#diff-7da68fc684eea654e797c077ff3c5bdc14a7f24f951280fb0b5f1e046bdef364R102). We'll also want to sync this change over to the [`configurable` docs](https://github.com/vectordotdev/vector/blob/master/src/sources/file.rs#L70-L73) as we continue to migrate off of writing cue directly.

Would you mind making these changes @aroberts?

**Comment by aroberts** [comment]
I updated the configurable docs.

Did you mean to link a different file? It's not clear what you mean by "same thing as here"

**Comment by aroberts** [comment]
ah, I see. I am pro-multiline, I was patterning off of nearby long lines. I committed your suggestions, thanks!

**Review by spencergilbert — APPROVED** [review]
❤️ Thanks!

**Comment by spencergilbert** [comment]
> ah, I see. I am pro-multiline, I was patterning off of nearby long lines. I committed your suggestions, thanks!

There's definitely some _odd_ formatting here and there, to be sure. We've been trying to clean things up as we see them 😄

**Comment by neuronull** [comment]
There appears to be a formatting error in the cue docs... Possibly a whitespace problem.

```
./scripts/check-docs.sh
Validating cue files formatting...
Incorrectly formatted CUE files

--- /dev/fd/63	2022-10-26 19:55:30.819717337 +0000
+++ /dev/fd/62	2022-10-26 19:55:30.819717337 +0000
@@ -41636,14 +41636,14 @@
 ^Iconfiguration: {$
 ^I^Iacknowledgements: configuration._source_acknowledgements$
 ^I^Iexclude: {$
-^I^I^Icommon:      false$
+^I^I^Icommon: false$
 ^I^I^Idescription: """$
 ^I^I^I^IArray of file patterns to exclude. [Globbing](#globbing) is supported.*Takes precedence over the [`include` option](#include).*$
 ^I^I^I^INote that the `exclude` patterns are applied _after_ Vector attempts to glob everything in `include`. That is, Vector will still$
 ^I^I^I^Itry to list all of the files matched by `include` and then filter them by the `exclude` patterns. This can be impactful if$
 ^I^I^I^I`include` includes directories with contents that vector does not have access to."$
 ^I^I^I^I"""$
-^I^I^Irequired:    false$
+^I^I^Irequired: false$
 ^I^I^Itype: array: {$
 ^I^I^I^Idefault: null$
 ^I^I^I^Iitems: type: string: {$
make: *** [Makefile:454: check-docs] Error 1
Error: Process completed with exit code 2.
```

File: website/cue/reference/components/sources/file.cue [hunk 1]
```diff
@@ -51,7 +51,12 @@ components: sources: file: {
 		acknowledgements: configuration._source_acknowledgements
 		exclude: {
 			common:      false
```
Comment (spencergilbert) [inline website/cue/reference/components/sources/file.cue:53]:
```suggestion
			common: false
```

File: website/cue/reference/components/sources/file.cue [hunk 2]
```diff
@@ -51,7 +51,12 @@ components: sources: file: {
...
-			description: "Array of file patterns to exclude. [Globbing](#globbing) is supported.*Takes precedence over the [`include` option](#include).*"
+			description: """
+				Array of file patterns to exclude. [Globbing](#globbing) is supported.*Takes precedence over the [`include` option](#include).*
+				Note that the `exclude` patterns are applied _after_ Vector attempts to glob everything in `include`. That is, Vector will still
+				try to list all of the files matched by `include` and then filter them by the `exclude` patterns. This can be impactful if
+				`include` includes directories with contents that vector does not have access to."
+				"""
 			required:    false
```
Comment (spencergilbert) [inline website/cue/reference/components/sources/file.cue:60]:
```suggestion
			required: false
```

**Review by spencergilbert — COMMENTED** [review]
Noting that this passes the cue check for me locally - but I expect these changes to work. Going to be very happy when we don't need to wrestle with cue anymore.

**Comment by aroberts** [comment]
I committed these suggestions, but I'll note that they're not lines that changed as part of my PR, so either the cue settings in the build pipeline changed, or cue has some crazy dependent parsing where the switch to multiline string affected downstream parsing. TBH this may be effort outsized to the outcome- judging from my initial google efforts, not many people have run into this problem, so I'm not sure how worth it is for either of us to fight with the build pipeline for more days.

**Comment by spencergilbert** [comment]
> I committed these suggestions, but I'll note that they're not lines that changed as part of my PR, so either the cue settings in the build pipeline changed, or cue has some crazy dependent parsing where the switch to multiline string affected downstream parsing. TBH this may be effort outsized to the outcome- judging from my initial google efforts, not many people have run into this problem, so I'm not sure how worth it is for either of us to fight with the build pipeline for more days.

Yeah, so if I recall correctly - cue will will try and match the spaces between key/values _most_ of the time:

```text
foo:      "bar"
fizzbuzz: "bazz"
```

Except, I guess if it's multiline? I'm _confident_ 😅 that this is the last round of golf to satisfy cue.

**Comment by aroberts** [comment]
Much appreciated - this sentence must be shared with the public!!!

**Comment by spencergilbert** [comment]
Success! Thanks again @aroberts!

**Pull request merged**